rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,90 @@
+---
+authors: Andrew LeFevre (andrew.lefevre@goteleport.com)
+state: draft
+---
+
+# RFD 98 - Registered OpenSSH Nodes
+
+## Required approvers
+
+* Engineering: @jakule && @r0mant
+* Product: @klizhentas
+* Security: @reedloden
+
+## What
+
+Allow OpenSSH nodes to be registered in a Teleport cluster.
+
+## Why
+
+[Agentless EC2 discovery mode](https://github.com/gravitational/teleport/issues/17865) will discover and configure OpenSSH nodes so they can authenticate with a cluster. But those OpenSSH nodes aren't registered as `node` resources in the backend. We need a way to register agentless OpenSSH nodes as `node` resources so they can be viewed and managed by users.
+
+## Details
+
+### Registering nodes
+
+OpenSSH nodes can already be registered using `tctl create -f /path/to/node.yml`, though some changes should be made to make the process more straightforward for users. Currently nodes cannot be created with `tctl`, only upserted. This limitation should be removed so users can create nodes without having to pass `-f` to `tctl create`.
+
+Furthermore, `tctl` should not require as many fields to be set when creating nodes. This is an example `node` resource that will work with `tctl create` today:
+
+```yaml
+kind: node
+metadata:
+  name: 5da56852-2adb-4540-a37c-80790203f6a9
+spec:
+  addr: 1.2.3.4:22
+  hostname: agentless-node
+version: v2
+```
+
+`metadata.name` should be auto-generated if it is not already set so user's don't have to generate GUIDs themselves.
+
+### RBAC
+
+When OpenSSH nodes are registered RBAC does not work as expected. Even when `auth_service.session_recording` is set to `proxy` in an Auth Server's config file, checking RBAC rules is not done correctly. RBAC logic will have to be updated to correctly handle OpenSSH nodes. Currently RBAC checks are done on Teleport agent nodes, unless the Proxy is in `proxy` session recording mode, then the Proxy handles all RBAC checks. To avoid overloading the Proxy when many nodes are registered, Teleport agent nodes should continue to handle RBAC checks whenever possible. The Proxy will still have to handle RBAC checks for registered OpenSSH nodes, so the question is how will registered OpenSSH nodes be identified vs. Teleport agent nodes?
+
+#### Detecting OpenSSH nodes
+
+##### Option 1 - add information to SSH host certs
+
+When SSH host certificates are created for OpenSSH nodes, we set add an extension `x-teleport-role` that specifies what Teleport component the certificate is for. We could add another role type for OpenSSH nodes, so that when a Proxy connects to an OpenSSH node and reviews it's certificate, it could see from the `x-teleport-role` extension that the node is a registered OpenSSH node and preform the RBAC check appropriately.
```

Comment:
```suggestion
When SSH host certificates are created for OpenSSH nodes, we set add an extension `x-teleport-role` that specifies what Teleport component the certificate is for. We could add another role type for OpenSSH nodes, so that when a Proxy connects to an OpenSSH node and reviews its certificate, it could see from the `x-teleport-role` extension that the node is a registered OpenSSH node and preform the RBAC check appropriately.
```

What would the value of `x-teleport-role` be in this example?

Reply:
Probably something like `openssh-node` or `registered-openssh-node`, but N/A due to my other comment regarding this option.